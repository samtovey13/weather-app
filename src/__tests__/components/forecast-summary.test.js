import React from 'react';
import { getByTestId, render } from "@testing-library/react";
import ForecastSummary from '../../components/forecast-summary';

describe("ForecastSummary", () => {

  it("renders correctly", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={1525046400000} 
        icon="211"
        description="Hazy" 
        temperature={10} 
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct date, icon, description and temperature props", () => {
    const { getByText } = render(
      <ForecastSummary 
        date={1525046400000} 
        icon="211"
        description="Hazy" 
        temperature={10} 
      />
    );

    expect(getByText("Mon 30th Apr")).toHaveClass("date");
    expect(getByText("Hazy")).toHaveClass("description");
    expect(getByText("10Â°c")).toHaveClass("temperature");
  });

  it("renders the correct icon", () => {
    const { getByTestId } = render(
      <ForecastSummary 
        date={1525046400000} 
        icon="211"
        description="Hazy" 
        temperature={10} 
      />
    );

    expect(getByTestId("icon-id")).toHaveClass("icon");
    expect(getByTestId("icon-id")).toContainElement(getByTestId("weather-icon-id"));
    expect(getByTestId("weather-icon-id")).toHaveAttribute("name", "owm");
  });
})

