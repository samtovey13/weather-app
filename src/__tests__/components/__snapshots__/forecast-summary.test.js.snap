// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary-card"
  >
    <span
      class="date"
      data-testid="date-id"
    >
      Mon 30th Apr
    </span>
    <span
      class="temperature"
      data-testid="temperature-id"
    >
      10Â°c
    </span>
    <span
      class="description"
      data-testid="description-id"
    >
      Hazy
    </span>
    <span
      class="icon"
      data-testid="icon-id"
    >
      <i
        class="wi wi-day-thunderstorm"
        data-testid="weather-icon-id"
        name="owm"
      />
    </span>
    <button>
      More Details
    </button>
  </div>
</DocumentFragment>
`;
